EDHM-UI: init at 3.0.46

Michael-K-Williams opened the pull request:
Add `edhm-ui` version 3.0.46 - Elite Dangerous HUD Mod Manager, a GUI application for managing HUD modifications in the Elite Dangerous space simulation game.

  **What the application does:**
  - Provides a user-friendly interface for installing and configuring HUD color themes
  - Manages HUD modifications for both Elite Dangerous: Horizons and Odyssey expansions
  - Includes preset HUD configurations and custom theme creation tools
  - Handles mod installation, backup, and restoration functionality

This package enables NixOS users to edit their HUD in Elite Dangerous more interactively, with a GUI, stopping them from needing to manually edit GraphicsConfiguration.xml within the game files.

  **Redistribution Permission:** The copyright holder Fred89210 has granted explicit permission for EDHM
   to be redistributed as part of the EDHM-UI package in Nixpkgs (see REDISTRIBUTION_EXCEPTION.md in the
   EDHM repository).

  ## Things done

  - Built on platform:
    - [x] x86_64-linux
  - Tested, as applicable:
    - [x] Ran `nixpkgs-review` on this PR
    - [x] Tested basic functionality of all binary files, usually in `./result/bin/`
  - [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md),
  [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md),
  [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other
  READMEs

  ---

  Add a 👍 [reaction](https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to [pull requests you find important](https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

  ---

  **Key changes from reviewer feedback:**
  - Uses `fetchzip` instead of `fetchurl` for better source handling
  - Uses modern `hash` instead of `sha256`
  - Implements `stdenv.mkDerivation (finalAttrs: {)` pattern
  - Uses `makeBinaryWrapper` and proper desktop integration
  - Selective file installation (excludes unnecessary system libraries)
  - Addresses all licensing concerns with explicit redistribution permission

Commit:
maintainers: add michael-k-williams

edhm-ui: init at 3.0.45

Elite Dangerous HUD Mod Manager - A tool for managing HUD modifications in Elite Dangerous.

EDHM UI is a GPL v3 licensed user interface for managing Elite Dangerous HUD modifications.
It includes EDHM (Elite Dangerous HUD Mod) which has a custom restrictive license.

Michael-K-Williams commented:
Removing withs having conversed with an acquaintance, fixing formatting errors.

Commit:
Changes:
- Making sourceRoot less ambiguous

RossSmyth reviewed (changes requested):
Definitely needs to clean up. But not too bad. Also please clean up the commits.

Michael-K-Williams commented:
> Please follow PR formatting guidelines.
> 
> The licensing of this is complex and I'm unsure how to reconcile it.
> 
>     1. Within their release archives they distribute Microsoft proprietary DLLs, probably patched for hooking I would guess, but still.
> 
>     2. https://github.com/psychicEgg/EDHM is non-redist and unfree, and this project is explicitly breaking that license. It is redistributing EDHM. There are no exceptions laid out in the non-redist licensing.
> 
>     3. The source of EDHM_UI itself is GPL3+
> 
>     4. It distributes a bunch of third-party binaries.

The main licensing of this is easy. 

1. Both EDHM Devs have given explicit permission for the creator of EDHMUI to use and redistribute their code in EDHMUI. EDHMUI has a GPLv3 License, meaning that I am allowed to redistribute it. I am redistributing it in a pre-compiled binary that they provide. In addition to this, I have the permission of the developer of EDHMUI to redistribute EDHMUI, not that I need it under GPLv3.

2. In regards to the Microsoft DLLs & Third-Party binaries used within it, I don't understand how those DLLs are not allowed(?) or somehow problematic(?)... 

In total, I don't see an issue here, could you elaborate?

Commit:
edhm-ui: init at 3.0.46

HUD modification manager for Elite Dangerous

Changes made to address reviewer feedback:
- Use fetchzip instead of fetchurl for better source handling
- Replace sha256 with hash following modern Nix conventions
- Use stdenv.mkDerivation (finalAttrs: {) pattern instead of rec
- Replace makeWrapper with makeBinaryWrapper
- Move buildInputs to runtimeDependencies for autopatchelfHook
- Use makeDesktopItem and copyDesktopItems for desktop integration
- Remove manual library path handling, let autopatchelfHook handle it
- Follow meta description guidelines (short, factual, no period)
- Move license descriptions to custom license attribute sets

RossSmyth commented on pkgs/by-name/ed/edhm-ui/package.nix:
```diff
@@ -0,0 +1,175 @@
+        shortName = "edhm-custom";
+        fullName = "EDHM Custom License";
+        url = null;
+        free = false;
```
It is also not redistributable

RossSmyth commented on pkgs/by-name/ed/edhm-ui/package.nix:
```diff
@@ -0,0 +1,175 @@
+    # Copy application resources
+    cp -r resources $out/opt/edhm-ui/
+
+    # Copy minimal locales (just English)
```
Being selective about locales is not a good idea though.

RossSmyth commented on pkgs/by-name/ed/edhm-ui/package.nix:
```diff
@@ -0,0 +1,175 @@
+    mkdir -p $out/opt/edhm-ui
+
+    # Copy main executable
+    cp edhm-ui-v3 $out/opt/edhm-ui/
+    chmod +x $out/opt/edhm-ui/edhm-ui-v3
```
`install`

RossSmyth commented on pkgs/by-name/ed/edhm-ui/package.nix:
```diff
@@ -0,0 +1,175 @@
+    cp *.pak *.bin *.dat snapshot_blob.bin v8_context_snapshot.bin $out/opt/edhm-ui/
+
+    # Copy required shared libraries (only essential ones)
+    cp libEGL.so libGLESv2.so libffmpeg.so $out/opt/edhm-ui/
```
Are these required/patched or can we just use the nixpkgs ones?

RossSmyth commented on pkgs/by-name/ed/edhm-ui/package.nix:
```diff
@@ -0,0 +1,175 @@
+    cp locales/en-US.pak $out/opt/edhm-ui/locales/
+
+    # Create wrapper script
+    makeBinaryWrapper $out/opt/edhm-ui/edhm-ui-v3 $out/bin/edhm-ui
```
Should just be a symlink, shouldn't need a wrapper anymore.

RossSmyth reviewed (changes requested):
Getting close.

1. Please follow the commit and PR title guidelines
2. Please remove unneeded function inputs
3. In general when doing something like `mkdir thing && cp source thing`, should just use install instead. Unless copying multiple items.

RossSmyth reviewed (changes requested):
Looks good, just a little clean-up and I would call it done. 

1. Please follow the PR title format
2. Clean-up the inputs

After that I think it's good. Once you clean it up can you post a screenshot of it working to "prove" the derivation works as expected?
